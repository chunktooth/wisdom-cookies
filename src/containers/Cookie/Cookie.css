.btn-shelf {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  flex-direction: row;
  position: relative;
  z-index: 5;
}

.cookie-container {
  width: 100%;
  height: 300px;
  overflow: hidden;
}

.left-cookie, .right-cookie {
  position: relative;
  cursor: pointer;
  transition: all .6s;
}

.left-cookie {
  top: 55px;
  left: 80px;
  width: 322px;
  z-index: 2;
  transform: rotate(-20deg);
}

.right-cookie {
  top: 47px;
  right: 99px;
  width: 280px;
  z-index: 1;
  -webkit-transform: rotate(-20deg);
  -ms-transform: rotate(-20deg);
  transform: rotate(0deg);
}

.left-cookie-broken {
  left: -125px;
  transform: rotate(-6deg);
  z-index: 2;
}

.right-cookie-broken {
  left: 100px;
  transform: rotate(-16deg);
  z-index: 2;
}

.wisdom-wrapper {
  position: relative;
  z-index: 5;
  bottom: 180px;
  left: 8px;
}

.message {
  position: relative;
  top: 60px;
  left: 188px;
  width: 320px;
  color: #3A4B77;
  font-size: .8em;
  font-weight: bold;
  margin: 0;
}

.put-inna-jar,
.eat-another-cookie {
  display: inline-block;
}

.eat-another-cookie {
  margin-left: 15px;
}

.to-jar,
.eat-cookie,
.wisdom-jar {
  cursor: pointer;
  box-shadow: 0px 0px 10px 2px #FFF;
}

.to-jar:hover,
.eat-cookie:hover,
.wisdom-jar:hover {
  box-shadow: 0px 0px 20px 2px #FFF;
}

.to-jar {
  width: 80px;
  padding-bottom: 4.5px;
}

.eat-cookie {
  width: 80px;
  padding: 10px 0;
}

.wisdom-jar {
  width: 70px;
  padding: 7px;
}

.btn-txt {
  margin: 0;
}

@media screen and (max-width: 601px) {
  .left-cookie {
    width: 300px;
  }

  .right-cookie {
    width: 260px;
  }
}

@media screen and (max-width: 559px) {
  .left-cookie {
    width: 260px;
  }

  .right-cookie {
    width: 220px;
  }

  @media screen and (max-width: 479px) {
    .left-cookie {
      width: 200px;
      left: 60px;
    }

    .right-cookie {
      width: 180px;
      right: 60px;
    }
  }
}