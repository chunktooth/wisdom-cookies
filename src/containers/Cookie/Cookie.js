import React, { Component } from 'react';
import { connect } from 'react-redux';
import { keepInJar } from '../../actions';
import leftCookie from '../../images/left-cookie.png';
import rightCookie from '../../images/right-cookie.png';
import wisdom from '../../images/wisdom-paper.jpg';
import toJar from '../../images/put-inna-jar.png';
import eatCookie from '../../images/eat-cookie.png';
import wisdomJar from '../../images/wisdom-jar.png';
import PropTypes from 'prop-types';
import './Cookie.css';

export class Cookie extends Component {
  constructor(props) {
    super(props);
    this.state = {
      revealWisdom: false,
      brokenCookie: false, 
      wisdom: {}
    };
  }

  eatCookie = () => {
    const { wisdoms } = this.props;

    this.setState({ 
      revealWisdom: !this.state.revealWisdom,
      brokenCookie: !this.state.brokenCookie
    });
    this.state.revealWisdom === true ? 
      this.hideWisdom() : 
      this.showWisdom(wisdoms);
  }; 

  showWisdom = async (wisdoms) => {
    const randomKey = Math.floor(Math.random() * wisdoms.length);
    const wisdom = wisdoms[randomKey];
    await this.setState({ wisdom });
  };

  hideWisdom = () => {
    this.setState({ 
      wisdom: {},
      brokenCookie: false,
      revealWisdom: false 
    });
  };

  putWisdomInJar = () => {
    let { jar } = this.props;
    let { wisdom } = this.state;

    if (!jar.find(wisdomInJar => wisdomInJar.id === 
      wisdom.id)) {
      this.props.keepInJar(wisdom);
    }
    this.setState({ 
      revealWisdom: false,
      brokenCookie: false 
    });
  }

  render() {
    let leftCookieState = this.state.brokenCookie ===
      true ? 'left-cookie-broken' : '';
    let rightCookieState = this.state.brokenCookie ===
      true ? 'right-cookie-broken' : '';

    return (
      <div className='Cookie'> 
        <div className='btn-shelf'>  
          <div className='jar-o-wisdoms'>
            <img src={wisdomJar}
              className='wisdom-jar' 
              alt="Revisit a jar of wisdoms" />
            <p className='btn-txt'>
              {`(${this.props.jar.length})
              Wisdom in Jar`}
            </p>
          </div>

          { this.state.revealWisdom &&
            <div className='jar-n-cookies'>
              <div className='put-inna-jar'>
                <img src={toJar}
                  className='to-jar'
                  alt="A jar to throw in wisdoms"
                  onClick={this.putWisdomInJar} />
                <p className='btn-txt'>Keep Wisdom</p>
              </div>

              <div className='eat-another-cookie'>
                <img src={eatCookie}
                  className='eat-cookie'
                  alt="Eat another cookie"
                  onClick={this.eatCookie} />
                <p className='btn-txt'>Eat Another!</p>
              </div>
            </div>
          } 
        </div>

        { !this.state.broken && 
          <div className='cookie-container'>
            <img src={leftCookie}
              className={`left-cookie ${leftCookieState}`}
              alt="left piece of cookie"
              onClick={this.eatCookie} />  
            <img src={rightCookie}
              className={`right-cookie ${rightCookieState}`}
              alt="right piece of cookie"
              onClick={this.eatCookie} />
          </div>
        }

        { this.state.revealWisdom &&
          <div className='wisdom-wrapper'>
            <img src={wisdom}
              className='wisdom-paper'
              alt="Wisdom on a paper" />
            <p className='message'>
              {this.state.wisdom.message}
            </p>
          </div> 
        }
      </div> 
    );
  }
}
 
export const mapStateToProps = (state) => {
  return {
    wisdoms: state.wisdoms,
    jar: state.jar
  };
};

export const mapDispatchToProps = (dispatch) => {
  return {
    keepInJar: (wisdom) => (dispatch(keepInJar(wisdom)))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Cookie);

Cookie.propTypes = {
  keepInJar: PropTypes.func,
  wisdoms: PropTypes.array,
  jar: PropTypes.array
};