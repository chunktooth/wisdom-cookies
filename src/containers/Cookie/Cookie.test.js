import React from 'react';
import { Cookie } from './Cookie';
import { shallow } from 'enzyme';

describe('Cookie', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Cookie />);
  });

  it.skip('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});