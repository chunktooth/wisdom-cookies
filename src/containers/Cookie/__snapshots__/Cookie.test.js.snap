// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookie should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cookie
    jar={
        Array [
            Object {
              "id": "5280",
              "message": "Mankind is unkind, man",
            },
            Object {
              "id": "1234",
              "message": "Godluck and Goodspeed!",
            },
          ]
    }
    keepInJar={[Function]}
    wisdoms={
        Array [
            Object {
              "id": "5280",
              "message": "Mankind is unkind, man..",
            },
            Object {
              "id": "1234",
              "message": "Godluck and Goodspeeeed!",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="cookie-container"
>
          <img
                    alt="left piece of cookie"
                    className="left-cookie "
                    onClick={[Function]}
                    src="left-cookie.png"
          />
          <img
                    alt="right piece of cookie"
                    className="right-cookie "
                    onClick={[Function]}
                    src="right-cookie.png"
          />
</div>,
        <div
          className="jar-o-wisdom"
>
          <img
                    alt="Revisit a jar of wisdoms"
                    className="wisdom-jar"
                    src="wisdom-jar.png"
          />
          <p
                    className="btn-txt"
          >
                    (2)
                                Wisdom in Jar
          </p>
</div>,
        false,
      ],
      "className": "Cookie",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="left piece of cookie"
              className="left-cookie "
              onClick={[Function]}
              src="left-cookie.png"
/>,
            <img
              alt="right piece of cookie"
              className="right-cookie "
              onClick={[Function]}
              src="right-cookie.png"
/>,
          ],
          "className": "cookie-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "left piece of cookie",
              "className": "left-cookie ",
              "onClick": [Function],
              "src": "left-cookie.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "right piece of cookie",
              "className": "right-cookie ",
              "onClick": [Function],
              "src": "right-cookie.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Revisit a jar of wisdoms"
              className="wisdom-jar"
              src="wisdom-jar.png"
/>,
            <p
              className="btn-txt"
>
              (2)
                          Wisdom in Jar
</p>,
          ],
          "className": "jar-o-wisdom",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Revisit a jar of wisdoms",
              "className": "wisdom-jar",
              "src": "wisdom-jar.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "(2)
            Wisdom in Jar",
              "className": "btn-txt",
            },
            "ref": null,
            "rendered": "(2)
            Wisdom in Jar",
            "type": "p",
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="cookie-container"
>
            <img
                        alt="left piece of cookie"
                        className="left-cookie "
                        onClick={[Function]}
                        src="left-cookie.png"
            />
            <img
                        alt="right piece of cookie"
                        className="right-cookie "
                        onClick={[Function]}
                        src="right-cookie.png"
            />
</div>,
          <div
            className="jar-o-wisdom"
>
            <img
                        alt="Revisit a jar of wisdoms"
                        className="wisdom-jar"
                        src="wisdom-jar.png"
            />
            <p
                        className="btn-txt"
            >
                        (2)
                                    Wisdom in Jar
            </p>
</div>,
          false,
        ],
        "className": "Cookie",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="left piece of cookie"
                className="left-cookie "
                onClick={[Function]}
                src="left-cookie.png"
/>,
              <img
                alt="right piece of cookie"
                className="right-cookie "
                onClick={[Function]}
                src="right-cookie.png"
/>,
            ],
            "className": "cookie-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "left piece of cookie",
                "className": "left-cookie ",
                "onClick": [Function],
                "src": "left-cookie.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "right piece of cookie",
                "className": "right-cookie ",
                "onClick": [Function],
                "src": "right-cookie.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Revisit a jar of wisdoms"
                className="wisdom-jar"
                src="wisdom-jar.png"
/>,
              <p
                className="btn-txt"
>
                (2)
                            Wisdom in Jar
</p>,
            ],
            "className": "jar-o-wisdom",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Revisit a jar of wisdoms",
                "className": "wisdom-jar",
                "src": "wisdom-jar.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "(2)
            Wisdom in Jar",
                "className": "btn-txt",
              },
              "ref": null,
              "rendered": "(2)
            Wisdom in Jar",
              "type": "p",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
